File: programs/libc/src/unistd/getgid.c
#include <unistd.h>
#include <user_syscall.h>

// The getgid() function shall return the real group ID of the calling
// process. The getgid() function shall not modify errno.

gid_t getgid(void)
{
	gid_t ret = _user_syscall(GETGID, 0);
	return ret;
}

File: rust_kernel/src/taskmaster/syscall/getgid.rs
//! sys_getgid()

use super::scheduler::SCHEDULER;
use super::SysResult;

/// The getgid() function shall return the real group ID of the
/// calling process. The getgid() function shall not modify errno.
pub fn sys_getgid() -> SysResult<u32> {
    Ok(unpreemptible_context!({
        SCHEDULER.lock().current_thread_group().credentials.gid
    }))
}
